File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.*;

class Solution {
    public int solution(String s) {
        int x = s.length();
        int answer = 0;
        for (int i = 0; i < x; i++) {
            if (check(s.substring(i, x) + s.substring(0, i))) answer++;
        }
        return answer;
    }
    
    public boolean check(String s) {
        Deque<Character> stack = new ArrayDeque<>();
        for (char ch : s.toCharArray()) {
            if (ch == '(' || ch == '[' || ch == '{') {
                stack.push(ch);
                continue;
            }
            if (stack.isEmpty()) return false;
            if (ch == ')' && stack.peek() == '(') {
                stack.pop();
            }
            if (ch == ']' && stack.peek() == '[') {
                stack.pop();
            }
            if (ch == '}' && stack.peek() == '{') {
                stack.pop();
            }
        }
        return stack.isEmpty()? true : false; 
    }
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def check(s):
    stack = []
    for c in s:
        if c in ['(', '[', '{']:
            stack.append(c)
            continue
        if not stack:
            return 0
        if c == ')' and stack[-1] == '(':
            stack.pop()
        if c == ']' and stack[-1] == '[':
            stack.pop()
        if c == '}' and stack[-1] == '{':
            stack.pop()
    return 0 if stack else 1

def solution(s):
    answer = 0
    for i in range(len(s)):
        if (check(s[i:] + s[:i])):
            answer += 1
    return answer